### DISCUSSION TITLE
WinRMOperationTimeoutError

### DISCUSSION THREAD

**Discussion by mfleurisson** [vmware/packer-examples-for-vsphere#912] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

latest

### VMware vSphere

8.0.2.00200

### HashiCorp Packer

1.10.3

### HashiCorp Packer Plugin for VMware vSphere

1.2.7

### Guest Operating System

ubuntu server 22.04

### Environment Details

ansible [core 2.15.0]
  config file = None
  configured module search path = ['/home/m.fleurisson@acecom.lan/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/m.fleurisson@acecom.lan/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/m.fleurisson@acecom.lan/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/m.fleurisson@acecom.lan/.local/bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True

pywinrm               0.4.3

### Description

when building windows server 2029,during the ansible task "Update the operating system and install additional packages" , it systematically terminates with the error 
"vsphere-iso.windows-server-standard-core: [WARNING]: ERROR DURING WINRM SEND INPUT - attempting to recover:
    vsphere-iso.windows-server-standard-core: WinRMOperationTimeoutError"

@

### Expected Behavior

build ok

### Actual Behavior

vsphere-iso.windows-server-standard-core: TASK [base : Update the operating system and install additional packages.] *****
    vsphere-iso.windows-server-standard-core: included: /home/m.fleurisson@acecom.lan/<sensitive>-examples-for-vsphere/<sensitive>/roles/base/tasks/windows.yml for default
    vsphere-iso.windows-server-standard-core: [WARNING]: ERROR DURING WINRM SEND INPUT - attempting to recover:
    vsphere-iso.windows-server-standard-core: WinRMOperationTimeoutError
Cancelling build after receiving interrupt
==> vsphere-iso.windows-server-standard-core: Step "StepProvision" failed
==> vsphere-iso.windows-server-standard-core: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?
    vsphere-iso.windows-server-standard-core:  [ERROR]: User interrupted execution
==> vsphere-iso.windows-server-standard-core: Power off VM...
==> vsphere-iso.windows-server-standard-core: NotAuthenticated
==> vsphere-iso.windows-server-standard-core: Destroying VM...
==> vsphere-iso.windows-server-standard-core: ServerFaultCode: The session is not authenticated.
==> vsphere-iso.windows-server-standard-core: Deleting cd_files image from remote datastore ...
    vsphere-iso.windows-server-standard-core: Closing sessions ....
Build 'vsphere-iso.windows-server-standard-core' errored after 29 minutes 37 seconds: context canceled

### Steps to Reproduce

reboot & relaunch

**Comment by tenthirtyam** [comment]
Hi there @mfleurisson - this hasn't been observed before in this project but take a look at the `winrm_timeout` option in the [plugin options](https://developer.hashicorp.com/packer/integrations/hashicorp/vsphere/latest/components/builder/vsphere-iso#optional-winrm-fields:). You might need to add this to the Windows build configuration.

**Reply by mfleurisson** [reply]
Hello,
The timeout is already set to 12h and the task crashes after about 30 minutes

**Reply by mfleurisson** [reply]
This seems to be a problem with the pywinrm python module : https://github.com/diyan/pywinrm/issues/323

Which pywinrm version should I use?

**Comment by mfleurisson** [comment, marked as answer]
This seems to be a known problem related to the ansible version, as described here https://github.com/ansible-collections/ansible.windows/issues/516 
Upgrading to ansible 2.15.5 seems to solve the problem

**Reply by tenthirtyam** [reply]
Please feel free to put in a pull request with an update to the requirements documentation.

Please follow the CONTRIBUTING.md guidlines. TY!

**Reply by tenthirtyam** [reply]
https://github.com/vmware-samples/packer-examples-for-vsphere/issues/917
